libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -209,9 +278,38 @@ void DataFlowAnalyzer::clearValues(set<YulString> _variables)
 		for (auto const& ref: m_references[name])
 			m_referencedBy[ref].erase(name);
 		m_references[name].clear();
+		// clear slot denoted by "name"
```

Comment:
This should not include the variables that are added in the first loop above - because the actual value of those variables does not change, only the value of their assigned expressions (if they are re-evaluated).